### PR TITLE
Typeahead with Pills (Rails)

### PR DESCRIPTION
**Pull request by thestephenmarshall**
#### Screens

Screen Shot 2020-10-07 at 12 33 07 PM

#### Breaking Changes

None

#### How to test this

- Navigate to `/kits/typeahead/rails` and test the doc examples for "with pills" examples listed
- Perform the same visual and functional smoke tests in Nitro Playbook dev docs

#### Checklist:

- [x] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `depreciation`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [x] **URGENCY** Please select a release milestone
- [x] **DEPLOY** Please add the `Milano` label when you are ready for a review.
- [x] **SCREENSHOT** Please add a screen shot or two.
- [x] **SPECS** Please cover your changes with specs.
- [x] **READ DOCS** Please make sure you have read and understand the [Playbook Release Process](https://github.com/powerhome/playbook/wiki/Playbook-Releases)

*The normal release cut off deadline is 3p EDT each week. Please reach out to the release team if you have an urgent request or need a release off cycle.*

### CODE REVIEW

File: app/pb_kits/playbook/pb_typeahead/typeahead.rb [hunk 1]
```diff
@@ -5,12 +5,19 @@ module PbTypeahead
...
       include Playbook::Props
 
+      prop :async, type: Playbook::Props::Boolean,
+                    default: false
+      prop :async_url
       prop :label
       prop :name
-      prop :value
+      prop :options, type: Playbook::Props::HashArray, default: []
+      prop :pills, type: Playbook::Props::Boolean,
```
Comment (thestephenmarshall) [inline app/pb_kits/playbook/pb_typeahead/typeahead.rb:14]: New props

File: app/pb_kits/playbook/pb_typeahead/typeahead.rb [hunk 2]
```diff
@@ -5,12 +5,19 @@ module PbTypeahead
     class Typeahead
       include Playbook::Props
 
+      prop :async, type: Playbook::Props::Boolean,
```
Comment (thestephenmarshall) [inline app/pb_kits/playbook/pb_typeahead/typeahead.rb:8]: New prop

File: app/pb_kits/playbook/pb_typeahead/typeahead.rb [hunk 3]
```diff
@@ -5,12 +5,19 @@ module PbTypeahead
...
+                    default: false
+      prop :async_url
```
Comment (thestephenmarshall) [inline app/pb_kits/playbook/pb_typeahead/typeahead.rb:10]: New props

File: app/pb_kits/playbook/pb_typeahead/typeahead.rb [hunk 4]
```diff
@@ -25,6 +32,22 @@ def data
...
+          placeholder: placeholder
+        }
+
+        base_options.merge!({
+          async: true,
+          asyncUrl: async_url
+        }) if async
+
+        base_options
+      end
```
Comment (thestephenmarshall) [inline app/pb_kits/playbook/pb_typeahead/typeahead.rb:50]: This provides the `props` hash for passing object `props` to the React kit instances

File: app/pb_kits/playbook/pb_typeahead/typeahead.rb [hunk 5]
```diff
@@ -5,12 +5,19 @@ module PbTypeahead
     class Typeahead
       include Playbook::Props
 
+      prop :async, type: Playbook::Props::Boolean,
+                    default: false
```
Comment (thestephenmarshall) [inline app/pb_kits/playbook/pb_typeahead/typeahead.rb:9]: I don't think this props really is necessary due to the fact that it should be implied that `load_options` means "async" inherently.

File: app/pb_kits/playbook/pb_typeahead/typeahead.rb [hunk 6]
```diff
@@ -5,12 +5,19 @@ module PbTypeahead
...
       prop :name
-      prop :value
+      prop :options, type: Playbook::Props::HashArray, default: []
+      prop :pills, type: Playbook::Props::Boolean,
+                    default: false
+
       prop :placeholder
       prop :search_term_minimum_length, default: 3
       prop :search_debounce_timeout, default: 250
+      prop :value
```
Comment (thestephenmarshall) [inline app/pb_kits/playbook/pb_typeahead/typeahead.rb:20]: Re-ordered

File: app/pb_kits/playbook/pb_typeahead/typeahead.rb [hunk 7]
```diff
@@ -5,12 +5,19 @@ module PbTypeahead
     class Typeahead
       include Playbook::Props
 
+      prop :async, type: Playbook::Props::Boolean,
+                    default: false
       prop :label
+      prop :load_options
```
Comment (thestephenmarshall) [inline app/pb_kits/playbook/pb_typeahead/typeahead.rb:11]: This is the *name or namespace* of the JavaScript function assigned to `window` which resolves a Promise, providing the XHR collected data from $anotherSource.

File: app/pb_kits/playbook/pb_typeahead/typeahead.rb [hunk 8]
```diff
@@ -25,6 +32,22 @@ def data
...
+          placeholder: placeholder
+        }
+
+        base_options.merge!({
+          async: true,
+          loadOptions: load_options,
+        }) if async
+
+        base_options
+      end
```
Comment (thestephenmarshall) [inline app/pb_kits/playbook/pb_typeahead/typeahead.rb:50]: This public method provides the `props` for the React version

File: app/pb_kits/playbook/pb_typeahead/_typeahead.jsx [hunk 9]
```diff
@@ -45,14 +51,27 @@ const Typeahead = (props: Props) => {
     },
     isClearable: true,
     isSearchable: true,
+    name,
     ...props,
   }
 
+  if (typeof(props.loadOptions) === 'string') selectProps.loadOptions = get(window, props.loadOptions)
```
Comment (thestephenmarshall) [inline app/pb_kits/playbook/pb_typeahead/_typeahead.jsx:58]: If we detect that we have a string here, we can assume that this means a variable in the `window` scope and that we should attempt to grab that and use it.

**Review by jasperfurniss — APPROVED** [review]

**Pull request merged**